### DISCUSSION TITLE
Update a value to "not" its current value

### DISCUSSION THREAD

**Discussion by derkoe** [diesel-rs/diesel#3325] [Ideas]
I want to create the following SQL statement:

```sql
UPDATE todos SET completed = NOT completed WHERE id = ?
```

When I do the following:
```rust
diesel::update(todos)
        .filter(id_col.eq(id))
        .set(completed_col.ne(completed_col))
        .execute(&conn)
```

I get the error 

```
135 |         .set(completed_col.ne(completed_col))
    |          --- ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `AsChangeset` is not implemented for `diesel::expression::grouped::Grouped<diesel::expression::operators::NotEq<axum_htmx_todos::schema::todos::completed, axum_htmx_todos::schema::todos::completed>>`
    |          |
    |          required by a bound introduced by this call
    |
    = help: the trait `AsChangeset` is implemented for `diesel::expression::grouped::Grouped<diesel::expression::operators::Eq<Left, Right>>`
```

It would be nice to have a `EqNot` operator for this.

**Comment by weiznich** [comment]
For future questions please try to include more information about your setup in your questions. Especially around your diesel version and the database backend you are using. There is no reason to introduce a new operator for this. What you tried would generate the following SQL:

```sql
UPDATE todos SET completed <> completed WHERE id = ?
```

which is just not valid SQL. Update statements require a assign via `=` and an expression for the right hand side. That's exactly the way this is implemented in diesel:

```rust
diesel::update(todos)
    .filter(id_col.eq(id))
    .set(completed_col.eq(diesel::dsl::not(completed_col))
    .execute(&conn)
```

**Comment by derkoe** [comment]
Thx, that was what I was looking for!

BTW: I'm using diesel 2.0.0 with PostgreSQL 14.
